src/microbe_stage/FluidCurrentDisplay.cs

```diff
@@ -0,0 +1,67 @@
+﻿using Godot;
+
+/// <summary>
+///   Displays fluid currents in the microbe stage
+/// </summary>
+public partial class FluidCurrentDisplay : GpuParticles3D
+{
+#pragma warning disable CA2213
+    private ShaderMaterial material = null!;
+#pragma warning restore CA2213
+
+    private StringName gameTimeParameterName = new("gameTime");
+    private StringName speedParameterName = new("speed");
+    private StringName chaoticnessParameterName = new("chaoticness");
+    private StringName scaleParameterName = new("scale");
+
+    private float time;
+
+    public override void _Ready()
+    {
+        base._Ready();
+
+        material = (ShaderMaterial)ProcessMaterial;
+    }
+
+    public override void _Process(double delta)
+    {
+        base._Process(delta);
+
+        GlobalPosition = new Vector3(GlobalPosition.X, 1.0f, GlobalPosition.Z);
+
+        if (!PauseManager.Instance.Paused)
+        {
+            time += (float)delta;
+
+            material.SetShaderParameter(gameTimeParameterName, time);
+
+            SpeedScale = 1.0f;
```

Comment:
This *might* need to be given the current game speed from the microbe world simulation as otherwise going to 2x mode might look a bit weird if a bunch of particles keep moving at the fixed speed while microbes speed up.